lib/bufref.c

```diff
@@ -116,12 +117,9 @@ CURLcode Curl_bufref_memdup(struct bufref *br, const void *ptr, size_t len)
   DEBUGASSERT(len <= CURL_MAX_INPUT_LENGTH);
 
   if(ptr) {
-    cpy = malloc(len + 1);
+    cpy = Curl_strndup(ptr, len);
```

Comment:
I think this is wrong: the function is supposed to duplicate the given byte count, but this will stop at the first null byte.
The original code behaves more like memdup, but appends a null byte as a sentinel in case the data is effectively a string.
Maybe we should have a such a generic dual-use duplication function.

Reply:
No, Curl_strndup() **does not** stop at the first null byte (any more). That is exactly what I mentioned above in this PR.

Reply:
My bad! I've been abused by the Curl_strndup() original code and the usage semantics associated with this name.
Sorry for the noise.

Reply:
I've been pondering on maybe renaming this function just to make it less likely that one would think that it has this functionality...

Reply:
> ... one would think that it has this functionality...

This happened to me: will probably affect others.